app/scripts/controllers/network/network-controller.test.js

```diff
@@ -50,224 +69,4106 @@ const PRE_1559_BLOCK = {
     '0x98bbdfbe1074bc3aa72a77a281f16d6ba7e723d68f15937d80954fb34d323369',
   uncles: [],
 };
-// Example block taken from Ethereum Mainnet post-EIP-1559
-const BLOCK = {
+
+/**
+ * A dummy block that matches the pre-EIP-1559 format (i.e. it has the
+ * `baseFeePerGas` property).
+ */
+const POST_1559_BLOCK = {
   ...PRE_1559_BLOCK,
   baseFeePerGas: '0x63c498a46',
 };
 
-const defaultControllerOptions = {
-  infuraProjectId: 'foo',
+/**
+ * An alias for `POST_1559_BLOCK`, for tests that don't care about which kind of
+ * block they're looking for.
+ */
+const BLOCK = POST_1559_BLOCK;
+
+/**
+ * A dummy value for the `projectId` option that `createInfuraClient` needs.
+ * (Infura should not be hit during tests, but just in case, this should not
+ * refer to a real project ID.)
+ */
+const DEFAULT_INFURA_PROJECT_ID = 'infura-project-id';
+
+/**
+ * Despite the signature of its constructor, NetworkController must take an
+ * Infura project ID. This object is mixed into the options first when a
+ * NetworkController is instantiated in tests.
+ */
+const DEFAULT_CONTROLLER_OPTIONS = {
+  infuraProjectId: DEFAULT_INFURA_PROJECT_ID,
 };
 
 /**
- * Builds a controller based on the given options, and calls the given function
- * with that controller.
- *
- * @param args - Either a function, or constructor options + a function. The
- * function will be called with the built controller.
- * @returns Whatever the callback returns.
+ * The set of networks that, when specified, create an Infura provider as
+ * opposed to a "standard" provider (one suited for a custom RPC endpoint).
  */
-async function withController(...args) {
-  const [constructorArgs, fn] = args.length === 2 ? args : [{}, args[0]];
-  const controller = new NetworkController({
-    ...defaultControllerOptions,
-    ...constructorArgs,
-  });
-  try {
-    return await fn({ controller });
-  } finally {
-    await controller.destroy();
-  }
-}
+const INFURA_NETWORKS = [
```

Comment:
Should we be using the values from the constants? (either consuming the values already exported, or assembling this array inside the constants file) https://github.com/MetaMask/metamask-extension/blob/develop/shared/constants/network.ts

Reply:
We could. Maybe this is just my opinion, but I don't like to reuse data that's been defined in implementation. If that data is incorrect, then the test will be incorrect too, but we wouldn't know about it because the tests may all pass. By defining this data in the test file, we can specify how we expect the world to look like, not what it actually is. Obviously we could get that wrong too, so it's not a bulletproof solution, but as it's isolated in this file we don't have to look so far to fix it. That was the thinking behind not using the pre-existing network constants.